dagshub/common/cli.py

```diff
@@ -49,5 +54,71 @@ def login(ctx, token, host):
         print("OAuth token added")
 
 
+def validate_repo(ctx, param, value):
+    parts = value.split("/")
+    if len(parts) != 2:
+        raise click.BadParameter("repo needs to be in the format <repo-owner>/<repo-name>")
+    return tuple(parts)
+
+
+def validate_user(ctx, param, value):
+    if value is None:
+        return None, None
+    parts = value.split(":")
+    if len(parts) != 2:
+        raise click.BadParameter("user needs to be in the format <username>:<password>")
+    return tuple(parts)
+
+
+def to_log_level(verbosity):
+    if verbosity == 0:
+        return logging.ERROR
+    elif verbosity == 1:
+        return logging.WARN
+    elif verbosity == 2:
+        return logging.INFO
+    elif verbosity >= 3:
+        return logging.DEBUG
+
+
+@cli.command()
+@click.argument("repo", callback=validate_repo)
+@click.argument("filename", type=click.Path(exists=True))
+@click.argument("target")
+@click.option("-m", "--message", help="Commit message for the upload")
+@click.option("-b", "--branch", help="Branch to upload the file to")
+@click.option("-u", "--user", callback=validate_user, help="Username and password in the format '<user>:<password>'."
+                                                           "This option is not recommended, instead leave this empty "
+                                                           "to use oauth or specify --token to use an existing "
+                                                           "user token.")
+@click.option("--update", is_flag=True, help="Specify --update to force update an existing file")
+@click.option("--token", help="Authenticate using an existing user token")
+@click.option("-v", "--verbose", default=0, count=True, help="Verbosity level")
+@click.pass_context
+def upload(ctx,
+           filename,
+           target,
+           repo,
+           message,
+           user,
+           branch,
+           token,
+           verbose,
+           update,
+           **kwargs):
+    """
+    Upload FILENAME to REPO at location TARGET.
+    REPO should be of the form <owner>/<repo-name>, i.e nirbarazida/yolov6.
+    TARGET should include the full path inside the repo, including the filename itself.
+    """
+    from dagshub.upload import Repo
+    dagshub.common.logging.logger.setLevel(to_log_level(verbose))
```

Comment:
Since this is a CLI, WDYT about doing `logging.BasicConfig()` here with the verbosity level?
That way we/user can debug requests and other used libraries, and not just seeing the logs from our library.

Reply:
We discussed this I think